### ISSUE TITLE
It should be possible to change the master password locally

### ISSUE THREAD

**Issue by devkid** [bitwarden/desktop#353]
This also applies for the browser extensions.

It should be possible to change the master password locally as the REST API allows to do this and there is no reason to redirect users from the local client to the Web UI of the vault installation to change the password.

As for the reasoning (putting on my tin foil hat): I can easily verify the version of locally installed clients (and potentially prevent updating them). However, I have no easy way to see if the source code of the Vault Web UI has been updated / potentially compromised (this applies for the official Bitwarden site as well as for self hosting).

**Comment by sectokia** [comment]
You have to remember the master-password is used for two purposes:
1-to decrypt/encrypt the vault
2-To authenticate to the server to store/retrieve encrypted copies of the vault.

What you are asking for would require a separation between the login password, and the vault password.

**Comment by devkid** [comment]
> What you are asking for would require a separation between the login password, and the vault password.
No, I'm not. I know that you need to send the new (encrypted) master password to the server. I just want to be able to initiate that from a local (trusted) client instead of from the Web UI, because I can't really verify if the Web UI has been tampered with and would log my new master password in clear text while entering it.
